### PR TITLE
Feature/prevent meta property serialisation

### PR DESCRIPTION
**Pull request by shamasis**
Refer to commit messages. In short: toJSON of base, no longer publishes meta properties

### CODE REVIEW

File: lib/collection/property-base.js [hunk 1]
```diff
@@ -97,21 +100,12 @@ _.assign(PropertyBase.prototype, /** @lends PropertyBase.prototype */ {
...
-            // Handle 'PropertyBase's
-            if (value && _.isFunction(value.toJSON)) {
-                accumulator[key] = value.toJSON();
-                return accumulator;
-            }
-
-            // Handle Strings
-            if (_.isString(value)) {
-                accumulator[key] = value;
+            if (PropertyBase.propertyIsMeta(value, key)) {
```
Comment (kamalaknn) [inline lib/collection/property-base.js]: Can we move this up to avoid plurality check for meta properties?

Comment (shamasis) [inline lib/collection/property-base.js]: Yes. Right. This should go up.

**Comment by shamasis** [comment]
@kamalaknn let this not get merged yet. Add your toJSON to request. 😞

**Comment by kamalaknn** [comment]
@shamasis Okay.

**Pull request closed without merging**